### DISCUSSION TITLE
Why is the data-test element for Username on the div instead of the input element?

### DISCUSSION THREAD

**Discussion by FrankAmsterdam** [cypress-io/cypress-realworld-app#1085] [General] [answered]
I'm having a lot of fun playing around with the Cypress RWA. Searching for the data-test attributes for username and password on the login page I expected them to be on the _input_ element but found them on a _div_ parent element. I noticed in my tests that it was possible to use these div to type in the text which suprises me a little bit.

**Comment by robertguss** [comment, marked as answer]
Hi @FrankAmsterdam this is because we are using [Material UI](https://material-ui.com/) components to render these inputs. You can see how they are being used here:
https://github.com/cypress-io/cypress-realworld-app/blob/86bc319b33a98fc70a5c2e431e98307f5f28e532/src/components/SignUpForm.tsx#L129
https://github.com/cypress-io/cypress-realworld-app/blob/86bc319b33a98fc70a5c2e431e98307f5f28e532/src/components/SignUpForm.tsx#L137
Hopefully that helps clarify things.

**Reply by FrankAmsterdam** [reply]
Does this mean that it is not possible to put the attribute on the input field in this way. Or does it just not matter to cypress that the attribute is on the div instead of the input field?

**Reply by robertguss** [reply]
Because we are using a 3rd party component library, there is only so much we have access to. As long as you can select the element you need with Cypress, there shouldn't be any issue.
